sdk/identity/azure_identity/src/client_certificate_credential.rs

```diff
@@ -66,28 +57,65 @@ impl Default for ClientCertificateCredentialOptions {
 #[derive(Debug)]
 pub struct ClientCertificateCredential {
     client_id: String,
-    certificate: Secret,
-    password: Secret,
+    key: PKey<Private>,
     endpoint: Url,
     pipeline: Pipeline,
-    send_certificate_chain: bool,
+    header: String,
     cache: TokenCache,
 }
 
 impl ClientCertificateCredential {
     /// Create a new `ClientCertificateCredential`.
-    pub fn new<C, P>(
+    pub fn new(
         tenant_id: String,
         client_id: String,
-        client_certificate: C,
-        client_certificate_password: P,
+        certificate: impl Into<Secret>,
         options: Option<ClientCertificateCredentialOptions>,
-    ) -> azure_core::Result<Arc<ClientCertificateCredential>>
-    where
-        C: Into<Secret>,
-        P: Into<Secret>,
-    {
+    ) -> azure_core::Result<Arc<ClientCertificateCredential>> {
+        validate_tenant_id(&tenant_id)?;
+        validate_not_empty(&client_id, "no client ID specified")?;
+
         let options = options.unwrap_or_default();
+
+        let cert_bytes = base64::decode(certificate.into().secret())
+            .with_context_fn(ErrorKind::Credential, || {
+                "failed to decode base64 certificate data"
+            })?;
+
+        let (key, cert, ca_chain) = parse_certificate(&cert_bytes, options.password.as_ref())?;
+        let thumbprint = cert
+            .digest(MessageDigest::sha1())
+            .with_context(ErrorKind::Credential, "failed to compute thumbprint")?
+            .to_vec();
+        let thumbprint = base64::encode(thumbprint);
+
+        #[cfg(test)]
+        let env = options.env.unwrap_or_default();
+        #[cfg(not(test))]
+        let env = Env::default();
+
+        let send_x5c = env
+            .var(AZURE_CLIENT_SEND_CERTIFICATE_CHAIN_ENV_KEY)
+            .map(|s| s == "1" || s.to_lowercase() == "true")
```

Comment:
Use `equal_ignore_ascii_case` or something like that so you're not allocating a new string just to compare. Equivalent to Go's `strings.EqualFold`.